### DISCUSSION TITLE
Historical issues

### DISCUSSION THREAD

**Discussion by paulk-asert** [apache/groovy-geb#229] [General]
I didn't notice before but geb issues were stored in the geb/issues repo not the geb/geb repo. The latter of these has been "moved" under Apache. Asciidoc is currently set up with a macro to point issue links to the geb/issues repo. I was thinking of just leaving the historical issues where they are for now and just have two macros for old and new issue links. We can always copy/move the geb/issues repo as well later if we want (and change the respective macro). Thoughts?

**Comment by paulk-asert** [comment]
I'll go ahead and merge that proposed PR. If folks aren't happy, we can rollback later.

**Comment by jonnybot0** [comment]
I do think we need to go through the old issues and migrate them to the new repository. I have read that Geb setup that separate repo as some means of keeping issue numbers that matched the old CodeHaus Jira system. I suppose that was to prevent needing to update old issues.

Bulk transfer is pretty straightforward using the [GitHub CLI](https://github.com/cli/cli). 

There are about 280 instances of the `issueh` macro. Seems like it should be a straightforward, if a little bit tedious script to write to go through the historical issues, transfer them to the new repository, and then update the `issueh` reference to simply use the new `issue` macro and the new number.

I used a code generator to bash out something that should do this:
```
#!/bin/bash

# Variables
SOURCE_REPO="source_owner/source_repo"
DEST_REPO="dest_owner/dest_repo"
CODE_DIR="/path/to/your/codebase"

# List all issues in the source repository
issue_numbers=$(gh issue list -R "$SOURCE_REPO" -s all -L 500 --json number -q '.[].number')

# Iterate over each issue number
for old_issue_number in $issue_numbers; do
    # Transfer the issue and capture the new issue URL
    new_issue_url=$(gh issue transfer "$old_issue_number" "https://github.com/$DEST_REPO" -R "$SOURCE_REPO" --json url -q '.url')

    # Extract the new issue number from the URL
    new_issue_number=$(echo "$new_issue_url" | grep -oP '(?<=/issues/)\d+')

    # Update references in the codebase
    if [ -n "$new_issue_number" ]; then
        echo "Updating references from issueh:$old_issue_number to issue:$new_issue_number"
        find "$CODE_DIR" -type f -exec sed -i '' "s/issueh:$old_issue_number/issue:$new_issue_number/g" {} +
    else
        echo "Failed to transfer issue #$old_issue_number"
    fi
done
```

Looks to me like it should work with a little testing. If there are no objections, I'll go ahead & do that a week from today (after spot-testing an issue or two).

**Comment by paulk-asert** [comment]
The main thing is that if you change the numbers, the issue mentions in existing commits will be lost/more confusing. There aren't many but for example this commit:

https://github.com/apache/groovy-geb/commit/54acf785209

The commit message is:

Add a failing test for geb/issues#395 by adding more test coverage for
delegation of waitFor().

Also, older commits use codehaus Jira references, e.g.

https://github.com/apache/groovy-geb/commit/be700038b3f

Has commit message:

GEB-287:Support for static atCheckWait on Page classes

Doesn't have an automatic link but at least you can get to:

https://github.com/geb/issues/issues/287

And it looks like there are lots of commits that use that convention.
